bolt-anything.bash

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+## STAGE 1 = build llvm-bolt
+## STAGE 2 = Instrument binary to run a workload with it to gather profiles for optimizing
+## STAGE 3 = Merging the created data to one file and remove the not needed data## Actually bug in llvm-bolt https://github.com/llvm/llvm-project/issues/56209
+## STAGE 4 = Optimize the binary with the created profile
+STAGE=
+
+## File or binary you want to instrument and then bolt
+BINARY=libLLVM-14.so
```

Comment:
a nice trick:

`: ${BINARY:=libLLVM-14.so}`

Then you can call the script like this and don't need to modify the source.
```
BINARY=myCustomLib.so ./bolt-anything.bash
```

The same applies to the other variables you declare here. It might improve ergonomics :)

Reply:
Yes, we may could add that here. Good point! 
Actually just sometimes the Binarypath is changing, it depends. For example gcc does need different PATH's.

Right now it seems to me bolting dynamic/shared binarys seems to break in general some things. The compiler works for some cases, but its still to unstable in my eyes. 

The bolt-anything.bash is anyways still good for a more dynamic usage of these scripts instead of only bolting llvm.

Reply:
if you want, you can add in this PR. but can be done later as well.

i would like to merge (+ squash again ;)) otherwise.